File: java/Mediator/src/CurrencyConverter.java


import java.util.HashMap;
import java.util.Map;

public class CurrencyConverter {
	Map<String, Float> conversionToDollarRates = new HashMap<String, Float>();;
	
	public CurrencyConverter(){
		this.conversionToDollarRates.put("USD", (float) 1);
		this.conversionToDollarRates.put("EUR", (float)0.9378);
		this.conversionToDollarRates.put("GBP", (float)0.7974);
		this.conversionToDollarRates.put("INR", (float)64.54);
	}

	public Map<String, Float> getConversionToDollarRates() {
		return conversionToDollarRates;
	}

	public void addNewCurrency(String currency, float rate){
		conversionToDollarRates.put(currency, rate);
	}
	
	public float convert(float amount, String currency){
		return amount/conversionToDollarRates.get(currency);
	}
	
}

File: ruby/Mediator/currency_converter.rb
class CurrencyConverter
  attr_accessor :conversionToDollarRates

  def initialize
    @conversionToDollarRates = {"USD" => 1, "EUR" => 0.9378, "GBP" => 0.7974, "INR" => 64.54}
  end

  def addNewCurrency(currency, rate)
    @conversionToDollarRates[currency] = rate
  end

  def convert(amount, currency)
    return amount/@conversionToDollarRates[currency]
  end
end